src/main/java/cms/gov/madie/measure/config/MeasureMongoConfig.java

```diff
@@ -47,4 +48,9 @@ public MongoConverter mongoConverter() throws Exception {
 
     return mongoConverter;
   }
+
+  @Bean
+  public GridFsTemplate gridFsTemplate(MongoDatabaseFactory mongoDbFactory) {
+    return new GridFsTemplate(mongoDbFactory, mongoConverter());
+  }
```

Comment:
GridFsOperations are being used. does that mean this bean in not required anymore?

Reply:
I'll check. The Spring Docs are unclear.
>The template can now be injected and used to perform storage and retrieval operations, as the following example shows: 

Doc proceeds to show examples using GridFsOperations and not Template...

See: https://docs.spring.io/spring-data/mongodb/reference/mongodb/template-gridfs.html

Reply:
okay, GridFsTemplate is an implementation of GridFsOperations same like MongoTemplate and MongoOperations, so, yeah its required. thanks for doc link!

Reply:
Ha! I failed to notice GridFsOperations was an interface.

Turns out, we don't need the explicit GridFsTemplate bean declaration. The Spring Mongo auto-configuration will create one if it is missing with default values:
```
  @Bean
  @ConditionalOnMissingBean({GridFsOperations.class})
  GridFsTemplate gridFsTemplate(MongoProperties properties, MongoDatabaseFactory factory, MongoTemplate mongoTemplate, MongoConnectionDetails connectionDetails) {
    return new GridFsTemplate(new GridFsMongoDatabaseFactory(factory, connectionDetails), mongoTemplate.getConverter(), connectionDetails.getGridFs() != null ? connectionDetails.getGridFs().getBucket() : null);
  }
  ```